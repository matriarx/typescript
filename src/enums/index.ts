export enum Environment {
  DEVELOPMENT = 'development',
  TEST = 'test',
  QA = 'qa',
  ALPHA = 'alpha',
  BETA = 'beta',
  UAT = 'uat',
  STAGE = 'stage',
  PRODUCTION = 'production',
  DEMO = 'demo',
}

export enum Path {
  INDEX = '/',
  AUTH = '/auth',
}

export default {
  Environment,
  Path,
}
