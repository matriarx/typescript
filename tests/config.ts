import {Environment} from 'enums'

describe('Config', () => {
  const ENV = process.env

  beforeEach(() => {
    jest.resetModules()
    process.env = {...ENV}
  })

  afterAll(() => {
    process.env = ENV
  })

  test('export default', async () => {
    const config = (await import('app/config')).default

    expect(config).toBeDefined()
    expect(config).toMatchSnapshot()
  })

  test('export config', async () => {
    const config = (await import('app/config')).config

    expect(config).toBeDefined()
  })

  test('NODE_ENV !== undefined', async () => {
    process.env['NODE_ENV'] = Environment.DEVELOPMENT

    const config = (await import('app/config')).default

    expect(config.environment).toStrictEqual(Environment.DEVELOPMENT)
  })

  test('NODE_ENV === undefined && ENVIRONMENT !== undefined', async () => {
    delete process.env['NODE_ENV']

    const config = (await import('app/config')).default

    expect(config.environment).toStrictEqual(Environment.DEVELOPMENT)
  })

  test('NODE_ENV === undefined && ENVIRONMENT === undefined', async () => {
    delete process.env['NODE_ENV']
    delete process.env['ENVIRONMENT']

    const config = (await import('app/config')).default

    expect(config.environment).toStrictEqual(Environment.DEVELOPMENT)
  })

  test('DEBUG !== undefined', async () => {
    process.env['DEBUG'] = 'true'

    const config = (await import('app/config')).default

    expect(config.debug).toStrictEqual(true)
  })

  test('DEBUG === undefined', async () => {
    delete process.env['DEBUG']

    const config = (await import('app/config')).default

    expect(config.debug).toStrictEqual(false)
  })

  test('HOST !== undefined', async () => {
    process.env['HOST'] = '0.0.0.0'

    const config = (await import('app/config')).default

    expect(config.host).toStrictEqual('0.0.0.0')
  })

  test('HOST === undefined', async () => {
    delete process.env['HOST']

    const config = (await import('app/config')).default

    expect(config.host).toStrictEqual('0.0.0.0')
  })

  test('PORT !== undefined', async () => {
    process.env['PORT'] = '80'

    const config = (await import('app/config')).default

    expect(config.port).toStrictEqual(80)
  })

  test('PORT === undefined', async () => {
    delete process.env['PORT']

    const config = (await import('app/config')).default

    expect(config.port).toStrictEqual(80)
  })
})
