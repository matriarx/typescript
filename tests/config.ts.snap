// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config export default 1`] = `
Object {
  "debug": true,
  "environment": "test",
  "host": "0.0.0.0",
  "port": 80,
}
`;
